public_html/administrator/components/com_jed/views/review/view.html.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @package    JED
+ *
+ * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Helper\ContentHelper;
+use Joomla\CMS\Language\Text;
+
+/**
+ * View to edit a review.
+ *
+ * @since  4.0.0
+ */
+class JedViewReview extends JViewLegacy
+{
+	protected $form;
+
+	/**
+	 * The item data.
+	 *
+	 * @var   object
+	 *
+	 * @since 4.0.0
+	 */
+	protected $item;
+
+	/**
+	 * The model state.
+	 *
+	 * @var   JObject
+	 *
+	 * @since 4.0.0
+	 */
+	protected $state;
+
+	/**
+	 * Display the view
+	 *
+	 * @param   string  $tpl  The name of the template file to parse; automatically searches through the template paths.
+	 *
+	 * @return  void
+	 *
+	 * @throws Exception
+	 *
+	 * @since 4.0.0
+	 */
+	public function display($tpl = null)
+	{
+		$this->state = $this->get('State');
+		$this->item  = $this->get('Item');
+		$this->form  = $this->get('Form');
+
+		// Check for errors.
+		if (count($errors = $this->get('Errors')))
+		{
+			throw new Exception(implode("\n", $errors), 500);
+		}
```

Comment:
This will throw notices on PHP 7.4 if you get a non-countable back. Preferred way is this:
```
$errors              = $model->getErrors();

		if ($errors && count($errors))
		{
			throw new RuntimeException(implode("\n", $errors), 500);
		}
```

Reply:
$errors              = $model->getErrors(); gives a deprecated warning?

Reply:
Yes correct. You don't see the deprecation with `->get()` because it is more or less a magic method. In J4 there should be no more JError and stuff, all should be exceptions although I don't know if they made it that far :) For now this is the code to use.